conan/external_tools/vswhere_test.py

```diff
@@ -0,0 +1,169 @@
+import os
+import unittest
+
+import nose
+
+from conans import tools
+from conans.errors import ConanException
+from conans.model.version import Version
+from conans import __version__ as client_version
+from conans.model import settings
+
+
+class vswhereTest(unittest.TestCase):
+
+    # Environment supossed:
+    #   - BuildTools 14 (2015)
+    #   - VS Community 14 (2015)
+    #
+    #   - BuildTools 15 (2017) OR VS Community 15 (2017)
+    modern_products = 1 # 2017 or higher versions -> vswhere()
+    all_modern_products = 2 # 2017 or higher versions -> vswhere(products=["*"])
+    modern_and_legacy_products = 2 # 2017 and lower versions -> vswhere(legacy=True)
+    only_legacy_products = 1
+    all_products = 3
+
+    def setUp(self):
+        if Version(client_version) < Version("1.1"):
+            raise nose.SkipTest('Only >= 1.1 version')
+
+    def vs_comntools_test(self):
+        # Fake path
+        print(tools.__file__)
+        with tools.environment_append({"VS150COMNTOOLS": "fake/path/here"}):
+            path = tools.vs_comntools("15")
+            self.assertEqual(path, "fake/path/here")
+
+        # VS 14 path
+        path = tools.vs_comntools("14")
+        self.assertEqual(path, "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\Common7\\Tools\\")
+
+        # VS 15 path (shouldn't be found as VS150COMNTOOLS is not set by default)
+        path = tools.vs_comntools("15")
+        self.assertEqual(path, None)
+
+    def vswhere_test(self):
+        # test there is some output
+        output = tools.vswhere()
+        self.assertIsNotNone(output)
+
+        # products and legacy not allowed
+        self.assertRaises(ConanException, tools.vswhere, products=["*"], legacy=True)
```

Comment:
This test could be moved to conan regular suite, because, is it actually doing something with the tool? or just internal validation?

Reply:
Just internal validation but I dont know if such a tiny test worths to be in the regular suite. The rest actually need "real" VS installations in the systems... Just tell me what you think

Reply:
If requires the visuals, then it is ok.